src/Sulu/Bundle/MediaBundle/Entity/CollectionRepository.php

```diff
@@ -389,4 +390,22 @@ public function setAccessControlQueryEnhancer(AccessControlQueryEnhancer $access
     {
         $this->accessControlQueryEnhancer = $accessControlQueryEnhancer;
     }
+
+    public function findDescendantIdsById($id)
+    {
+        $queryBuilder = $this->createQueryBuilder('subCollection')
+            ->select('subCollection.id')
+            ->leftJoin($this->_entityName, 'collection', Join::WITH, 'collection.id = :id')
```

Comment:
Should this not be a inner join as it need to exist and if I remember correctly inner joins are faster then left joins?

Reply:
I think the confusing part was the misusage of a `join` in general... This is not a join in the general sense, but a cartesian product, whereby the elimination happens only based on the `where` clause happening later. Rewrote it a bit, to make that part more clear.